File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s081988820_comments_removed.go
// Correct Go translation of the Java code
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	solve()
}

func solve() {
	sc := bufio.NewScanner(os.Stdin)
	var H, W int
	if sc.Scan() {
		parts := strings.Fields(sc.Text())
		H, _ = strconv.Atoi(parts[0])
		W, _ = strconv.Atoi(parts[1])
	}

	c := make([][]int, 10)
	for i := range c {
		c[i] = make([]int, 10)
	}
	for i := 0; i < 10; i++ {
		sc.Scan()
		parts := strings.Fields(sc.Text())
		for j := 蹈 0; j < 10; j++ {
			c[i][j], _ = strconv.Atoi(parts[j])
		}
	}

	min := make([]int, 10)
	for i := 0; i < 10; i++ {
		min[i] = c[i][1]
	}

	for tc := 0; tc < 10; tc++ {
		for i := 0; i < 10; i++ {
			for j := 0; j < 10; j++ {
				temp := c[i][j] + min[j]
				if temp < min[i] {
					min[i] = temp
				}
			}
		}
	}

	ans := 0
	for h := 0; h < H; h++ {
		sc.Scan()
		parts := strings.Fields(sc.Text())
		for _, part := range parts {
			A, _ := strconv.Atoi(part)
			if A >= 0 {
				ans += min[A]
			}
		}
	}
	fmt.Println(ans)
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C/s081988820_comments_removed.c
#include <stdio.h>

int main() {
    int H, W;
    int c[10][10];
    int min[10];
    int ans = 0;
    int i, j, tc, h, w, A;

    scanf("%d %d", &H, &W);

    for (i = 0; i < 10; i++) {
        for (j = 0; j < 10; j++) {
            scanf("%d", &c[i][j]);
        }
    }

    for (i = 0; i < 10; i++) {
        min[i] = c[i][1];
    }

    for (tc = 0; tc < 10; tc++) {
        for (i = 0; i < 10; i++) {
            for (j = 0; j < 10; j++) {
                if (c[i][j] + min[j] < min[i]) {
                    min[i] = c[i][j] + min[j];
                }
            }
        }
    }

    for (h = 0; h < H; h++) {
        for (w = 0; w < W; w++) {
            scanf("%d", &A);
            if (A >= 0) {
                ans += min[A];
            }
        }
    }

    printf("%d\n", ans);
}
// End of Code
